ethpm_types/manifest.py

```diff
@@ -165,11 +165,12 @@ def check_both_version_and_name(cls, values):
 
     @root_validator
     def check_contract_source_ids(cls, values):
-        if values.get("contract_types") is not None and values.get("sources") is not None:
-            for alias in values["contract_types"]:
-                source_id = values["contract_types"][alias].source_id
-                if source_id and (source_id not in values["sources"]):
-                    raise ValueError(f"'{source_id}' missing from `sources`.")
+        contract_types = values.get("contract_types", {}) or {}
```

Comment:
```suggestion
        contract_types = values.get("contract_types", {})
```

Reply:
If this change is made, tests fail (for good reason). The failure is because it is `None` so the second arg to `get()` is not hit and it tried to iterate through `None`, which is a builtin failure.

If you look at the type of `contract_types`, it is allowed to be `None`.
Also, because of how the `Field` is configured, it _will_ always be `None` when not present in the model.

So really, `values.get("contract_types") or {}` is needed. The optional second arg is the one that could potentially be removed but i feel better with it there.